asv_bench/benchmarks/categoricals.py

```diff
@@ -1,114 +1,140 @@
     goal_time = 0.2
 
     def setup(self):
-        N = 100000
-        self.s = pd.Series((list('aabbcd') * N)).astype('category')
+        N = 10**5
+        self.s = pd.Series(list('aabbcd') * N).astype('category')
+
+        self.a = pd.Categorical(list('aabbcd') * N)
+        self.b = pd.Categorical(list('bbcdjk') * N)
+
+    def time_concat(self):
+        pd.concat([self.s, self.s])
+
+    def time_union(self):
+        union_categoricals([self.a, self.b])
+
 
-        self.a = pd.Categorical((list('aabbcd') * N))
-        self.b = pd.Categorical((list('bbcdjk') * N))
+class Constructor(object):
 
+    goal_time = 0.2
+
+    def setup(self):
+        N = 10**5
         self.categories = list('abcde')
-        self.cat_idx = Index(self.categories)
+        self.cat_idx = pd.Index(self.categories)
         self.values = np.tile(self.categories, N)
         self.codes = np.tile(range(len(self.categories)), N)
 
-        self.datetimes = pd.Series(pd.date_range(
-            '1995-01-01 00:00:00', periods=10000, freq='s'))
+        self.datetimes = pd.Series(pd.date_range('1995-01-01 00:00:00',
+                                                 periods=N / 10,
+                                                 freq='s'))
+        self.datetimes_with_nat = self.datetimes.copy()
+        self.datetimes_with_nat.iloc[-1] = pd.NaT
 
         self.values_some_nan = list(np.tile(self.categories + [np.nan], N))
         self.values_all_nan = [np.nan] * len(self.values)
 
-    def time_concat(self):
-        concat([self.s, self.s])
-
-    def time_union(self):
-        union_categoricals([self.a, self.b])
-
     def time_constructor_regular(self):
-        Categorical(self.values, self.categories)
+        pd.Categorical(self.values, self.categories)
 
     def time_constructor_fastpath(self):
-        Categorical(self.codes, self.cat_idx, fastpath=True)
+        pd.Categorical(self.codes, self.cat_idx, fastpath=True)
 
     def time_constructor_datetimes(self):
-        Categorical(self.datetimes)
+        pd.Categorical(self.datetimes)
 
     def time_constructor_datetimes_with_nat(self):
-        t = self.datetimes
-        t.iloc[-1] = pd.NaT
-        Categorical(t)
+        pd.Categorical(self.datetimes_with_nat)
 
     def time_constructor_with_nan(self):
-        Categorical(self.values_some_nan)
+        pd.Categorical(self.values_some_nan)
 
     def time_constructor_all_nan(self):
-        Categorical(self.values_all_nan)
+        pd.Categorical(self.values_all_nan)
 
 
-class Categoricals2(object):
+class ValueCounts(object):
+
     goal_time = 0.2
 
-    def setup(self):
-        n = 500000
+    params = [True, False]
+    param_names = ['dropna']
+
+    def setup(self, dropna):
+        n = 5 * 10**5
         np.random.seed(2718281)
         arr = ['s%04d' % i for i in np.random.randint(0, n // 10, size=n)]
-        self.ts = Series(arr).astype('category')
+        self.ts = pd.Series(arr).astype('category')
+        self.dropna = dropna
```

Comment:
This is not needed I think? You can directly use the arg `dropna` in the time method